clang/lib/CConv/RewriteUtils.cpp

```diff
@@ -1522,7 +1573,26 @@ void RewriteConsumer::HandleTranslationUnit(ASTContext &Context) {
         std::string newTy = getStorageQualifierString(D) +
                             PV->mkString(Info.getConstraints().getVariables()) +
                             ABRewriter.getBoundsString(D);
-        RewriteThese.insert(DAndReplace(D, DS, newTy));
+
+        VarDecl *VD = dyn_cast<VarDecl>(D);
+        if (VD != nullptr) {
+          if (VD->getType().getTypePtr()->getAs<TypedefType>() == nullptr) {
+            RewriteThese.insert(DAndReplace(D, DS, newTy));
+          } else {
+            bool isTypedefStruct = false;
+            for (const auto name : Info.TypedefStructNames) {
+              if (name == PV->getOriginalTy()) {
```

Comment:
You can use built-in search functionality provided by `std::set` i.e., `Info.TypedefStructNames.find` to simplify this. No need to use linear search.